### PR TITLE
Add gitea push hook

### PR DESCRIPTION
**Pull request by SuperSandro2000**
The gitea webhook payload is slightly different and not compatible with the github endpoint. This PR adds a new api endpoint `/api/push-gitea` which understands the gitea webhook to trigger jobsets evaluation.

### CODE REVIEW

File: src/lib/Hydra/Controller/API.pm [hunk 1]
```diff
@@ -285,6 +285,22 @@ sub push_github : Chained('api') PathPart('push-github') Args(0) {
     $c->response->body("");
 }
 
+sub push_gitea : Chained('api') PathPart('push-gitea') Args(0) {
+    my ($self, $c) = @_;
+
+    $c->{stash}->{json}->{jobsetsTriggered} = [];
+
+    my $in = $c->request->{data};
+    my $url = $in->{repository}->{clone_url} or die;
```
Comment (DarkKirb) [inline src/lib/Hydra/Controller/API.pm:294]:
```suggestion
    my $url = $in->{repository}->{clone_url} or die;
    $url =~ s/.git$//;
```

**Review by DarkKirb — APPROVED** [review]
This PR works for me, but it would be nice if `push_gitea` removed the .git suffix from the repo URL, because I haven’t included them in my gitea flake inputs

File: src/lib/Hydra/Controller/API.pm (hunk 1 above)
Comment (SuperSandro2000) [reply src/lib/Hydra/Controller/API.pm:294]: Would that still match if the flake input had ``.git``?

Comment (DarkKirb) [reply src/lib/Hydra/Controller/API.pm:294]: i would think so, since the flake/url matching is done with LIKE and wildcards before and after the URL

**Comment by vtrf** [comment]
Hello, is this a work in progress? I wanted to use this feature on my Gitea instance and if needed, test it.

**Comment by SuperSandro2000** [comment]
This PR is done and deployed in production.

**Comment by CyborgPotato** [comment]
Can confirm that this PR works in my deployment as well :+1:

**Comment by RTUnreal** [comment]
is there a update when this will be upstreamed? Or will this PR be a patch user need to always apply?

**Comment by icewind1991** [comment]
I had to modify it to strip the `https://` prefix from the url to make it work with `git+ssh` job sources but works fine otherwise

**Comment by SuperSandro2000** [comment]
@icewind1991 can you make a suggestion with that change? Then i'll add it.

@Ericson2314 could you review/merge this?

File: src/lib/Hydra/Controller/API.pm [hunk 2]
```diff
@@ -285,6 +285,23 @@ sub push_github : Chained('api') PathPart('push-github') Args(0) {
...
+    $url =~ s/.git$//;
```
Comment (icewind1991) [inline src/lib/Hydra/Controller/API.pm:295]:
```suggestion
    $url =~ s/.git$//;
    $url =~ s/^https?:\/\///;
```

Disclaimer: this is probably the 10th or so line of perl I've ever written :see_no_evil:

Comment (Ericson2314) [reply src/lib/Hydra/Controller/API.pm:295]: We can follow-up for this part.

Comment (gador) [reply src/lib/Hydra/Controller/API.pm:295]: @icewind1991 how did you apply your change? Do you use a custom fork of hydra?

**Pull request merged**